### DISCUSSION TITLE
how do u safely cancel a search

### DISCUSSION THREAD

**Discussion by christmasjoke** [WerWolv/ImHex#989] [Get Help] [answered]
i tried a regex with .+ in a big exe file and it's been running for a little over a day, trying to close the program says it's waiting for the search to finish, if i alt+f4 does it damage the exe file in any way?

**Comment by WerWolv** [comment, marked as answer]
Oh god lol
I improved that behaviour in one of the latest commits now. In the footer of the main window there's a progress bar when you start a search. Right next to it is a button to cancel the operation. That's the proper way to do it if it works.

Killing ImHex is in most cases fine since it only rarely writes data to disk. It definitely shouldn't damage anything
